appserver/extras/docker-images/server-full/src/main/docker/Dockerfile

```diff
@@ -0,0 +1,93 @@
+RUN true \
+    && apt-get update \
+    && apt-get install -y wget unzip coreutils gpg \
+    && rm -rf /var/lib/apt/lists/* \
+    && groupadd --gid 1000 payara \
+    && useradd -u 1000 -M -s /bin/bash -d ${HOME_DIR} payara -g payara \
+    && echo payara:payara | chpasswd \
+    && mkdir -p ${DEPLOY_DIR} \
+    && mkdir -p ${CONFIG_DIR} \
+    && mkdir -p ${SCRIPT_DIR} \
+    && chown -R payara: ${HOME_DIR} \
+    && true
+
+
+# Install tini as minimized init system
+RUN true \
+    && wget --no-verbose -O /tini https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini \
+    && wget --no-verbose -O /tini.asc https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini.asc \
+    && gpg --batch --keyserver "hkp://p80.pool.sks-keyservers.net:80" --recv-keys 595E85A6B1B4779EA4DAAEC70B588DFF0527A9B7 \
+    && gpg --batch --verify /tini.asc /tini \
+    && chmod +x /tini \
+    && true
+
+USER payara
+WORKDIR ${HOME_DIR}
+
+# Download and unzip the Payara distribution
+COPY --chown=payara:payara ${PAYARA_UNPACKED} ${HOME_DIR}
+
+RUN true \
+    && mv ${HOME_DIR}/payara*/ ${PAYARA_DIR} \
+    # Configure the password file for configuring Payara
+    && echo "AS_ADMIN_PASSWORD=\nAS_ADMIN_NEWPASSWORD=${ADMIN_PASSWORD}" > /tmp/tmpfile \
+    && echo "AS_ADMIN_PASSWORD=${ADMIN_PASSWORD}" >> ${PASSWORD_FILE} \
+    # Configure the payara domain
+    && ${PAYARA_DIR}/bin/asadmin --user ${ADMIN_USER} --passwordfile=/tmp/tmpfile change-admin-password --domain_name=${DOMAIN_NAME} \
+    && ${PAYARA_DIR}/bin/asadmin --user=${ADMIN_USER} --passwordfile=${PASSWORD_FILE} start-domain ${DOMAIN_NAME} \
+    && ${PAYARA_DIR}/bin/asadmin --user=${ADMIN_USER} --passwordfile=${PASSWORD_FILE} enable-secure-admin \
+    && for MEMORY_JVM_OPTION in $(${PAYARA_DIR}/bin/asadmin --user=${ADMIN_USER} --passwordfile=${PASSWORD_FILE} list-jvm-options | grep "Xm[sx]"); do\
+        ${PAYARA_DIR}/bin/asadmin --user=${ADMIN_USER} --passwordfile=${PASSWORD_FILE} delete-jvm-options $MEMORY_JVM_OPTION;\
+    done \
+    # FIXME: when upgrading this container to Java 10+, this needs to be changed to '-XX:+UseContainerSupport' and '-XX:MaxRAMPercentage'
+    && ${PAYARA_DIR}/bin/asadmin --user=${ADMIN_USER} --passwordfile=${PASSWORD_FILE} create-jvm-options '-XX\:+UnlockExperimentalVMOptions:-XX\:+IgnoreUnrecognizedVMOptions:-XX\:+UseCGroupMemoryLimitForHeap:-XX\:MaxRAMFraction=1' \
+    && ${PAYARA_DIR}/bin/asadmin --user=${ADMIN_USER} --passwordfile=${PASSWORD_FILE} set-log-attributes com.sun.enterprise.server.logging.GFFileHandler.logtoFile=false \
+    && ${PAYARA_DIR}/bin/asadmin --user=${ADMIN_USER} --passwordfile=${PASSWORD_FILE} stop-domain ${DOMAIN_NAME} \
+    # Cleanup unused files
+    && rm -rf \
+        /tmp/tmpFile \
+        payara.zip \
+        ${PAYARA_DIR}/glassfish/domains/${DOMAIN_NAME}/osgi-cache \
+        ${PAYARA_DIR}/glassfish/domains/${DOMAIN_NAME}/logs \
+        ${PAYARA_DIR}/glassfish/domains/domain1 \
```

Comment:
domain name hardcoded

Reply:
It is correct - this is a list of files and directories which should be removed.

Reply:
But when a Docker image is created with domain1, this will remove the domain we actually need.

Reply:
By either copying the domain1 or production domain into the image, another 3MB can be gained. (by filtering it out in the maven-resources-plugin execution)
Probably can do this later.

Reply:
Good idea!

Reply:
It does not create domains, it only uses domain production which is already in zip file dependency together with domain1 which is not needed. 
But to filter out the domain1 even before running docker plugin is good idea.
Second thing is that tiny installation - it could be somehow merged from it's own image OR installed by system command instead dlownloading it by wget which must be too installed first ... I don't know how much we can spare with that (and if) ...
But filter out the domain should be simple :)

Reply:
>Second thing is that tiny installation - it could be somehow merged from it's own image OR installed by system command instead dlownloading it by wget which must be too installed first ..

It depends on what the tiny Image is based. Bit since wget and install (and gpg verification) is in one RUN command, and thus in one layer, you can't optimize it more than it already is today.

Reply:
I moved filtering to maven phase - server-full is 3 MBs lighter, server-node is 6 MB lighter.